tests/test_interfaces.py

```diff
@@ -420,3 +420,223 @@ def test_output_format(
+            data_mapping=model_fixture.data_mapping,
+            num_iterations=torch.tensor(5),
+        )
+
+
+# Tests to assert that incorrect input to sample is caught
+@pytest.mark.parametrize("model_fixture", MODELS)
+@pytest.mark.parametrize("end_time", END_TIMES)
+@pytest.mark.parametrize("logging_step_size", LOGGING_STEP_SIZES)
+def test_positive_int_sample(model_fixture, end_time, logging_step_size):
+    # Assert that providing a positive int does NOT raise an error
+    sample(
+        model_fixture.url,
+        end_time,
+        logging_step_size,
+        num_samples=10,
+    )
+
+
+@pytest.mark.parametrize("model_fixture", MODELS)
+@pytest.mark.parametrize("end_time", END_TIMES)
+@pytest.mark.parametrize("logging_step_size", LOGGING_STEP_SIZES)
+def test_float_sample(model_fixture, end_time, logging_step_size):
+    # Assert that providing a float raises a ValueError
+    with pytest.raises(ValueError):
+        sample(
+            model_fixture.url,
+            end_time,
+            logging_step_size,
+            num_samples=10.5,
+        )
+
+
+@pytest.mark.parametrize("model_fixture", MODELS)
+@pytest.mark.parametrize("end_time", END_TIMES)
+@pytest.mark.parametrize("logging_step_size", LOGGING_STEP_SIZES)
+def test_negative_int_sample(model_fixture, end_time, logging_step_size):
+    # Assert that providing a float raises a ValueError
+    with pytest.raises(ValueError):
+        sample(
+            model_fixture.url,
+            end_time,
+            logging_step_size,
+            num_samples=-10,
+        )
+
+
+@pytest.mark.parametrize("model_fixture", MODELS)
+@pytest.mark.parametrize("end_time", END_TIMES)
+@pytest.mark.parametrize("logging_step_size", LOGGING_STEP_SIZES)
+def test_zero_sample(model_fixture, end_time, logging_step_size):
+    # Assert that providing a float raises a ValueError
+    with pytest.raises(ValueError):
+        sample(
+            model_fixture.url,
+            end_time,
+            logging_step_size,
+            num_samples=0,
+        )
+
+
+@pytest.mark.parametrize("model_fixture", MODELS)
+@pytest.mark.parametrize("end_time", END_TIMES)
+@pytest.mark.parametrize("logging_step_size", LOGGING_STEP_SIZES)
+def test_tensor_int_sample(model_fixture, end_time, logging_step_size):
+    # Assert that providing a float raises a ValueError
+    with pytest.raises(ValueError):
+        sample(
+            model_fixture.url,
+            end_time,
+            logging_step_size,
+            num_samples=torch.tensor(10),
+        )
+
+
+# Tests to assert that incorrect input to ensemble_sample is caught
+@pytest.mark.parametrize("model_fixture", MODELS)
+@pytest.mark.parametrize("end_time", END_TIMES)
+@pytest.mark.parametrize("logging_step_size", LOGGING_STEP_SIZES)
+def test_positive_int_ensemble_sample(model_fixture, end_time, logging_step_size):
+    # Assert that providing a positive int does NOT raise an error
+    solution_mapping = [{"I": "I"}]
```

Comment:
This is moot given by comment above about consolidating `sample` and `ensemble_sample` methods using `pytest.mark.parametrize`, but as general feedback we want to avoid using model-specific content in model-agnostic tests. In other words, this line assumes that we'll have variables "I" in every model in our tests, which is not true now, nor do we expect it to be true in the future.

Reply:
Got it, thanks!